web_app/ingest_page/row_entry.py

```diff
@@ -86,16 +85,41 @@ def submit(self):
 
         self.clear_entry()
 
-        for row in user_entries:
-            for i, (key, value) in enumerate(row.items()):
-                label = customtkinter.CTkLabel(self, text=value)
-                label.grid(row=self.row_index, column=i, sticky='w', padx=(0, 5), pady=1)
+        ref = user_entries[0]['ref']
+
+        self.rows[ref] = {
+                'label_rows': [],
+                'container': customtkinter.CTkFrame(self)
+                }
+        container = self.rows[ref]['container']
+        self._container_config(container)
+        container.grid(row=self.row_index, column=0, columnspan=len(self.fields), sticky="ew", pady=5)
+        for i, row in enumerate(user_entries):
+            self.rows[ref]['label_rows'].append(self._draw_label_row(container, row, i))
             self.row_index += 1
 
         if self.on_enter is not None:
             self.on_enter(user_entries)
 
+    def _draw_label_row(self, parent, row, grid_row):
```

Comment:
I am guessing all of this additional code is for that grouped row view in which that is fine. But maybe consider extracting it as its own component.